firefox-ios/Client/Frontend/Library/Bookmarks/Edit Bookmark/EditBookmarkViewModel.swift

```diff
@@ -95,8 +99,21 @@ class EditBookmarkViewModel {
     func saveBookmark() {
         guard let selectedFolder, let node else { return }
         Task { @MainActor [weak self] in
-            _ = await self?.bookmarksSaver.save(bookmark: node,
-                                                parentFolderGUID: selectedFolder.guid)
+            let result = await self?.bookmarksSaver.save(bookmark: node,
+                                                         parentFolderGUID: selectedFolder.guid)
+            if selectedFolder.guid != self?.parentFolder.guid {
+                switch result {
+                case .success(let guid):
+                    if guid == nil {
```

Comment:
I don't understand this part. If the guid is nil it means we are updating the folder, so only in the case we save the recent Bookmark folder ? Also i'd use a guard here 
```swift
guard guid == nil else { return }
save.
```

Reply:
Correct - but I think the check is redundant since we only update bookmarks from this view, not create them, so it isn't actually doing anything. Good catch, will remove.